### PR TITLE
Configfiles

### PR DESCRIPTION
**Pull request by degenerated1123**
Cli-flags can now be written to a config file. Currently, specifying an option on the CL has higher priority than the one in the config-file. 

If you want a flag to appear inside the config-file, you need to specify a default value and the regarding config-section. These have been added to the constructor of `Cli::Flag`.

Config-files are saved with the current settings from the CL-args at the end of `main`, so, when you close the game manually.

### CODE REVIEW

File: src/utils/cli.cpp [hunk 1]
```diff
@@ -124,3 +222,49 @@ void ::Cli::printHelp()
...
+    // Dump using specified indentation. This is important for comments, since they use the same indentation!
+    std::string dumped = j.dump(JSON_DUMP_INDENTATION);
+
+    // Another pass, to insert the comments
+    for (Flag* f : Global::getFlagList())
+        dumped = f->documentConfigText(dumped);
+
+    // Write that one to disk now. Ensure the userdata-folder exists first, though
+    std::string userdata = Utils::getUserDataLocation();
+    Utils::mkdir(userdata);
```
Comment (astillich) [inline src/utils/cli.cpp]: Should check for error and return, the current impl will say that it failed to create the file and not the directory which is a bit misleading. Just a nitpick, though.

File: src/utils/cli.h [hunk 2]
```diff
@@ -23,13 +23,13 @@ namespace Cli
          * @param verboseFlag Verbose version of the short-flag given beforehand, with eg. "./app --do-foo",
          *             this would have to be "do-foo".
          * @param nparams number of parameters of the flag
-         * @param defaultValue default values of this flag
+         * @param defaultValue default values of this flag. For booleans (nparams==0), a "0" or "1" is used.
```
Comment (astillich) [inline src/utils/cli.h]: Terminology: arg vs. param ;)

File: src/utils/cli.h [hunk 3]
```diff
@@ -20,8 +23,16 @@ namespace Cli
          * @param verboseFlag Verbose version of the short-flag given beforehand, with eg. "./app --do-foo",
          *             this would have to be "do-foo".
          * @param nparams number of parameters of the flag
```
Comment (astillich) [inline src/utils/cli.h]: Terminology: should not mix args / params.

File: src/utils/cli.h [hunk 4]
```diff
@@ -40,15 +51,36 @@ namespace Cli
         std::vector<std::string> extractFlag();
 
         /**
-         * @return Arguments specified on the commandline of this flag
+         * @param Index of the argument to get
+         * @return Arguments specified on the commandline of this flag. Empty string if invalid index.
+         */
+        const std::string& getArgs(size_t i);
```
Comment (astillich) [inline src/utils/cli.h]: Should use unsigned: nparams (nargs) is unsigned too

File: src/utils/cli.h (hunk 3 above)
Comment (degenerated1123) [inline src/utils/cli.h]: Will rename everything to params, since that is what they are now. Good catch!

File: src/utils/cli.h (hunk 4 above)
Comment (degenerated1123) [inline src/utils/cli.h]: This function did something else before I implemented config-loading. Will change!

**Pull request merged**